kv/memberlist/mergeable.go

```diff
@@ -8,6 +8,7 @@ type Mergeable interface {
 	// Merge with other value in place. Returns change, that can be sent to other clients.
 	// If merge doesn't result in any change, returns nil.
 	// Error can be returned if merging with given 'other' value is not possible.
+	// This method may also modify the other parameter.
```

Comment:
Nit: Interface comment should describe "requirements" on implementations. The way it is phrased, it sounds like describing one specific implementation.

Furthermore random modifications of 'other' may break requirements described below. I would therefore remove this comment completely.

Reply:
That is exactly what the comment means, "may" indicates "is permitted to". Given it is an interface, I think it is clear that it is _not_ describing an implementation. I am happy to re-word it it it makes it easier to grok e.g.

>  	// Implementors of this method are permitted to modify the `other` parameter.

Reply:
Then let's also explain why:

```suggestion
// Implementors of this method are permitted to modify the other parameter, as memberlist-based KV store will not use the same "other" parameter in multiple Merge calls.
```